plugins/bufferlines/navic.nix

```
@@ -0,0 +1,121 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.navic;
+  helpers = import ../helpers.nix {inherit lib;};
+  mkListStr = helpers.defaultNullOpts.mkNullable (types.listOf types.str);
+in {
+  options.plugins.navic =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "nvim-navic";
+
+      package = helpers.mkPackageOption "nvim-navic" pkgs.vimPlugins.nvim-navic;
+
+      icons =
+        mapAttrs
+        (
+          name: default:
+            helpers.defaultNullOpts.mkStr default "icon for ${name}."
+        )
+        {
+          File = "󰈙 ";
+          Module = " ";
+          Namespace = "󰌗 ";
+          Package = " ";
+          Class = "󰌗 ";
+          Method = "󰆧 ";
+          Property = " ";
+          Field = " ";
+          Constructor = " ";
+          Enum = "󰕘";
+          Interface = "󰕘";
+          Function = "󰊕 ";
+          Variable = "󰆧 ";
+          Constant = "󰏿 ";
+          String = "󰀬 ";
+          Number = "󰎠 ";
+          Boolean = "◩ ";
+          Array = "󰅪 ";
+          Object = "󰅩 ";
+          Key = "󰌋 ";
+          Null = "󰟢 ";
+          EnumMember = " ";
+          Struct = "󰌗 ";
+          Event = " ";
+          Operator = "󰆕 ";
+          TypeParameter = "󰊄 ";
+        };
+
+      lsp = {
+        autoAttach = helpers.defaultNullOpts.mkBool false ''
+          Enable to have nvim-navic automatically attach to every LSP for current buffer. Its disabled by default.
+        '';
+
+        preference = helpers.defaultNullOpts.mkListStr "[]" ''
```

```suggestion
        preference = helpers.defaultNullOpts.mkNullable (with types; listOf str) "[]" ''
``` 
This helper doesn't exist